examples/example-xds/src/main/java/io/grpc/examples/helloworldxds/XdsHelloWorldClient.java

```diff
@@ -0,0 +1,107 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.examples.helloworldxds;
+
+import io.grpc.Channel;
+import io.grpc.ChannelCredentials;
+import io.grpc.Grpc;
+import io.grpc.InsecureChannelCredentials;
+import io.grpc.ManagedChannel;
+import io.grpc.StatusRuntimeException;
+import io.grpc.examples.helloworld.GreeterGrpc;
+import io.grpc.examples.helloworld.HelloReply;
+import io.grpc.examples.helloworld.HelloRequest;
+import io.grpc.xds.XdsChannelCredentials;
+import java.util.Arrays;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * A simple xDS client that requests a greeting from {@code HelloWorldServer} or {@link
+ * XdsHelloWorldServer}.
+ */
+public class XdsHelloWorldClient {
+  private static final Logger logger = Logger.getLogger(XdsHelloWorldClient.class.getName());
+
+  private final GreeterGrpc.GreeterBlockingStub blockingStub;
+
+  /** Construct client for accessing HelloWorld server using the existing channel. */
+  public XdsHelloWorldClient(Channel channel) {
+    blockingStub = GreeterGrpc.newBlockingStub(channel);
+  }
+
+  /** Say hello to server. */
+  public void greet(String name) {
+    logger.info("Will try to greet " + name + " ...");
+    HelloRequest request = HelloRequest.newBuilder().setName(name).build();
+    HelloReply response;
+    try {
+      response = blockingStub.sayHello(request);
+    } catch (StatusRuntimeException e) {
+      logger.log(Level.WARNING, "RPC failed: {0}", e.getStatus());
+      return;
+    }
+    logger.info("Greeting: " + response.getMessage());
+  }
+
+  /**
+   * Greet server. If provided, the first element of {@code args} is the name to use in the
+   * greeting. The second argument is the target server. A {@code --secure} flag is also accepted.
+   */
+  public static void main(String[] args) throws Exception {
+    String user = "xds world";
+    // The example defaults to the same behavior as the hello world example. To enable xDS, pass an
+    // "xds:"-prefixed string as the target.
+    String target = "localhost:50051";
+    ChannelCredentials credentials = InsecureChannelCredentials.create();
+    if (args.length > 0) {
+      if ("--help".equals(args[0])) {
+        System.out.println("Usage: [--secure] [NAME [TARGET]]\n");
+        System.err.println("  --secure  Use credentials provided by xDS. Defaults to insecure");
+        System.err.println("  NAME      The name you wish to be greeted by. Defaults to " + user);
+        System.err.println("  TARGET    The server to connect to. Defaults to " + target);
+        System.exit(1);
+      } else if ("--secure".equals(args[0])) {
```

Comment:
I am okay with `--secure` but I thought you wanted to change it to something more appropriate (because of the fallback to plaintext)

Reply:
I wanted to make that change in a separate PR.